ITensorGaussianMPS/src/gmps.jl

```diff
@@ -251,12 +580,51 @@ function ITensors.ITensor(u::Givens, s1::Index, s2::Index)
   return itensor(U, s2', s1', dag(s2), dag(s1))
 end
 
+function ITensors.ITensor(u::Givens, s1::Index, s2::Index, is_boguliobov::Bool)
+  if is_boguliobov
+    U = [
+      u.c 0 0 conj(u.s)
+      0 1 0 0
+      0 0 1 0
+      -(u.s) 0 0 u.c
+    ]
+    return itensor(U, s2', s1', dag(s2), dag(s1))
+  else
+    U = ITensor(u, s1, s2)
+  end
+end
+
 function ITensors.ITensor(sites::Vector{<:Index}, u::Givens)
   s1 = sites[u.i1]
   s2 = sites[u.i2]
   return ITensor(u, s1, s2)
 end
 
+function ITensors.ITensor(sites::Vector{<:Index}, u::Givens, is_boguliubov::Bool)
+  s1 = sites[u.i1]
+  s2 = sites[u.i2]
+  return ITensor(u, s1, s2, is_boguliobov)
+end
+
+function to_manybody_gates(s::Vector{<:Index}, C::Pairing)
+  is_bog = g -> abs(g.i2 - g.i1) == 2 ? false : true
+  s1 = g -> div(g.i1 - 1, 2) + 1
+  s2 = g -> div(g.i2 - 1, 2) + 1
+  U = [
+    ITensor(g, s[s1(g)], s[s2(g)], is_bog(g)) for
```

Comment:
Could we make use of the `ConservesNfParity` and `ConservesNf` wrappers here? Then we can dispatch on them with:
```julia
ITensor(::ConservesNfParity{<:Givens}, [...]) # Boguliobov rotation
ITensor(::ConservesNf{<:Givens}, [...]) # Number conserving rotation
```

In ITensors.jl the closest function to `to_manybody_gates` is called `ops`, i.e. https://github.com/ITensor/ITensors.jl/blob/v0.3.23/examples/gate_evolution/quantum_simulator.jl. We could also just call it `itensors`, since it constructs a list of ITensors.